packages/common/src/grpc-operation.js

```diff
@@ -0,0 +1,206 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module common/grpcOperation
+ */
+
+'use strict';
+
+var events = require('events');
+var modelo = require('modelo');
+
+/**
+ * @type {module:common/grpcService}
+ * @private
+ */
+var GrpcService = require('./grpc-service.js');
+
+/**
+ * @type {module:common/grpcServiceObject}
+ * @private
+ */
+var GrpcServiceObject = require('./grpc-service-object.js');
+
+/**
+ * An Operation object allows you to interact with APIs that take longer to
+ * process things.
+ *
+ * @constructor
+ * @alias module:common/grpcOperation
+ *
+ * @param {module:common/grpcService|module:common/grpcServiceObject} parent - The
+ *     parent object. This should be configured to use the longrunning.operation
+ *     service.
+ * @param {string} name - The operation name.
+ */
+function GrpcOperation(parent, name) {
+  var methods = {
+
+    /**
+     * Deletes an operation.
+     */
+    delete: {
+      protoOpts: {
+        service: 'Operations',
+        method: 'deleteOperation'
+      },
+      reqOpts: {
+        name: name
+      }
+    },
+
+    /**
+     * Checks to see if an operation exists.
+     */
+    exists: true,
+
+    /**
+     * Retrieves the operation.
+     */
+    get: true,
+
+    /**
+     * Retrieves metadata for the operation.
+     */
+    getMetadata: {
+      protoOpts: {
+        service: 'Operations',
+        method: 'getOperation'
+      },
+      reqOpts: {
+        name: name
+      }
+    }
+  };
+
+  var config = {
+    parent: parent,
+    id: name,
+    methods: methods
+  };
+
+  GrpcServiceObject.call(this, config);
+  events.EventEmitter.call(this);
+
+  this.completeListeners = 0;
+  this.hasActiveListeners = false;
+
+  this.listenForEvents_();
+}
+
+modelo.inherits(GrpcOperation, GrpcServiceObject, events.EventEmitter);
+
+/**
+ * Status text used to describe the current state of the operation.
+ *
+ * @private
+ */
+GrpcOperation.status = {
+  running: 'RUNNING',
+  done: 'DONE'
+};
+
+/**
+ * Cancel the operation.
+ *
+ * @param {function=} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this
+ *     request.
+ * @param {object} callback.apiResponse - The full API response.
+ */
+GrpcOperation.prototype.cancel = function(callback) {
+  var protoOpts = {
+    service: 'Operations',
+    method: 'cancelOperation'
+  };
+
+  var reqOpts = {
+    name: this.id
+  };
+
+  this.request(protoOpts, reqOpts, callback);
```

Comment:
I think we need a `callback = callback || common.util.noop` to make it optional here.